vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryIndexingOperatorDescriptor.java

```diff
@@ -0,0 +1,212 @@
+    private final String START_TAG = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n";
+    private final String hdfsConf;
+    private final Map<String, NodeControllerInfo> nodeControllerInfos;
+
+    public VXQueryIndexingOperatorDescriptor(IOperatorDescriptorRegistry spec, VXQueryIndexingDataSource ds,
+            RecordDescriptor rDesc, String hdfsConf, Map<String, NodeControllerInfo> nodeControllerInfos) {
+        super(spec, 1, 1);
+        collectionPartitions = ds.getCollectionPartitions();
+        indexPartitions = ds.getIndexPartitions();
+        dataSourceId = (short) ds.getDataSourceId();
+        totalDataSources = (short) ds.getTotalDataSources();
+        recordDescriptors[0] = rDesc;
+        this.childSeq = ds.getChildSeq();
+        this.tag = ds.getTag();
+        this.hdfsConf = hdfsConf;
+        this.nodeControllerInfos = nodeControllerInfos;
+    }
+
+    @Override
+    public IOperatorNodePushable createPushRuntime(IHyracksTaskContext ctx,
+            IRecordDescriptorProvider recordDescProvider, int partition, int nPartitions) throws HyracksDataException {
+        final FrameTupleAccessor fta = new FrameTupleAccessor(
+                recordDescProvider.getInputRecordDescriptor(getActivityId(), 0));
+        final int fieldOutputCount = recordDescProvider.getOutputRecordDescriptor(getActivityId(), 0).getFieldCount();
+        final IFrame frame = new VSizeFrame(ctx);
+        final IFrameFieldAppender appender = new FrameFixedFieldTupleAppender(fieldOutputCount);
+        final short partitionId = (short) ctx.getTaskAttemptId().getTaskId().getPartition();
+        final ITreeNodeIdProvider nodeIdProvider = new TreeNodeIdProvider(partitionId, dataSourceId, totalDataSources);
+        final String nodeId = ctx.getJobletContext().getApplicationContext().getNodeId();
+        final DynamicContext dCtx = (DynamicContext) ctx.getJobletContext().getGlobalJobData();
+
+        final String collectionName = collectionPartitions[partition % collectionPartitions.length];
+        final String indexName = indexPartitions[partition % indexPartitions.length];
+        final XMLParser parser = new XMLParser(false, nodeIdProvider, nodeId, appender, childSeq,
+                dCtx.getStaticContext());
+
+        return new AbstractUnaryInputUnaryOutputOperatorNodePushable() {
+            @Override
+            public void open() throws HyracksDataException {
+                appender.reset(frame, true);
+                writer.open();
+                hdfs = new HDFSFunctions(nodeControllerInfos, hdfsConf);
+            }
+
+            @Override
+            public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
+                fta.reset(buffer);
+                String collectionModifiedName = collectionName.replace("${nodeId}", nodeId);
+                String indexModifiedName = indexName.replace("${nodeId}", nodeId);
+                if (!collectionModifiedName.contains("hdfs:/")) {
```

Comment:
throw HyracksDataException with a message about not creating indexes on HDFS files.